core/domain/src/main/java/org/the_chance/honeymart/domain/usecase/ValidationUseCase.kt

```diff
@@ -0,0 +1,64 @@
+package org.the_chance.honeymart.domain.usecase
+
+import org.the_chance.honeymart.domain.util.ValidationState
+import java.util.regex.Pattern
+import javax.inject.Inject
+
+class ValidationUseCase @Inject constructor() {
```

Comment:
I do not know if this class holds more than one responsibility where the validation is done to the field of email and password in the same use case.

Reply:
bareq said usecases in same or similar screen meaning (user story) should be in one use case refrenece last  review